diff -Naur libkipi-15.12.2/src/interface.h libkipi-15.12.2.tpg/src/interface.h
--- libkipi-15.12.2/src/interface.h	2016-01-22 12:34:22.000000000 +0000
+++ libkipi-15.12.2.tpg/src/interface.h	2016-03-04 00:23:50.327044676 +0000
@@ -177,15 +177,7 @@
      */
     virtual bool loadRawPreview(const QUrl& url, QImage& image) = 0;
 
-    /** To perform demosiacing. Can be cancelled.
-     */
-    virtual bool decodeRawImage(const QUrl& url, QByteArray& imageData, int& width, int& height, int& rgbmax) = 0;
-
-    /** To cancel decodeRawImage() processing.
-     */
-    virtual void cancel() = 0;
-
-    /** Rteun true if url is a Raw file, else false
+    /** Return true if url is a Raw file, else false
      */
     virtual bool isRawFile(const QUrl& url) = 0;
 
diff -Naur libkipi-15.12.2/tests/common/kipiinterface.cpp libkipi-15.12.2.tpg/tests/common/kipiinterface.cpp
--- libkipi-15.12.2/tests/common/kipiinterface.cpp	2016-01-22 12:34:22.000000000 +0000
+++ libkipi-15.12.2.tpg/tests/common/kipiinterface.cpp	2016-03-04 00:28:22.324339345 +0000
@@ -242,17 +242,6 @@
         return m_decoder.loadRawPreview(image, url.toLocalFile());
     }
 
-    bool decodeRawImage(const QUrl& url, QByteArray& imageData, int& width, int& height, int& rgbmax)
-    {
-        return m_decoder.decodeRAWImage(url.toLocalFile(), KDcrawIface::RawDecodingSettings(),
-                                        imageData, width, height, rgbmax);
-    }
-
-    void cancel()
-    {
-        m_decoder.cancel();
-    }
-
     bool isRawFile(const QUrl& url)
     {
         QString   rawFilesExt = QLatin1String(m_decoder.rawFiles());
